drivers/vmwarevsphere/cloudinit.go

```diff
@@ -228,59 +230,118 @@ func (d *Driver) mountCloudInitIso(vm *object.VirtualMachine, dc *object.Datacen
 	return vm.AddDevice(d.getCtx(), add...)
 }
 
-func addSSHUserToYaml(yamlcontent, user, group, sshkey string) (string, error) {
-	cf := make(map[interface{}]interface{})
-	if err := yaml.Unmarshal([]byte(yamlcontent), &cf); err != nil {
-		return "", err
-	}
+func (d *Driver) addSSHUserToYaml(yamlcontent, user, group, sshkey string) (string, error) {
+	switch {
+	case strings.ToLower(d.MachineOS) == defaultMachineOS:
```

Comment:
i think you should explicitly check "linux" and do default cases to error out in case someone passes in a weird os. thoughts? i think you do this logic to check default machine os a bunch, maybe they should all be "linux" instead to say this code only happens if linux so it's clear

Reply:
done, using "linux" and "windows" as the cases instead and converting d.OS to lowercase elsewhere instead